File: 0100-same-tree/0100-same-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if(p == null || q==null){
            if(p==null && q ==null)return true;
            else return false;
        }
        if(p.val != q.val)return false;

        boolean l = isSameTree(p.left, q.left);
        boolean r = isSameTree(p.right, q.right);

        if(l && r)return true;
        else return false;
    }
}

File: 0100-same-tree/0100-same-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        flag = True
        def inorder(p, q):
            nonlocal flag
            if p is None and q is None:
                return 
            if p is None or q is None:
                flag = False
                return
            if p.val != q.val:
                flag = False

            inorder(p.left, q.left)
            inorder(p.right, q.right)
        inorder(p, q)
        return flag
        